Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp

```diff
@@ -34,6 +34,80 @@ namespace AZ::Render
     }
 
+    // Function for providing data required for debug drawing colliders
+    Physics::CharacterPhysicsDebugDraw::NodeDebugDrawData GetNodeDebugDrawData(
+        const Physics::CharacterColliderNodeConfiguration& colliderNodeConfig,
+        const EMotionFX::ActorInstance* instance,
+        const AZStd::unordered_set<size_t>* cachedSelectedJointIndices)
+    {
+        Physics::CharacterPhysicsDebugDraw::NodeDebugDrawData nodeDebugDrawData;
+        const EMotionFX::Actor* actor = instance->GetActor();
+        const EMotionFX::Node* joint = actor->GetSkeleton()->FindNodeByName(colliderNodeConfig.m_name.c_str());
+        if (!joint)
+        {
+            nodeDebugDrawData.m_valid = false;
+            return nodeDebugDrawData;
+        }
+
+        const size_t nodeIndex = joint->GetNodeIndex();
+        nodeDebugDrawData.m_selected = cachedSelectedJointIndices &&
+            (cachedSelectedJointIndices->empty() || cachedSelectedJointIndices->find(nodeIndex) != cachedSelectedJointIndices->end());
+
+        const EMotionFX::Transform& actorInstanceGlobalTransform = instance->GetWorldSpaceTransform();
+        const EMotionFX::Transform& emfxNodeGlobalTransform =
+            instance->GetTransformData()->GetCurrentPose()->GetModelSpaceTransform(nodeIndex);
+        nodeDebugDrawData.m_worldTransform = (emfxNodeGlobalTransform * actorInstanceGlobalTransform).ToAZTransform();
+        nodeDebugDrawData.m_valid = true;
+        return nodeDebugDrawData;
+    };
+
+    // Function for proviuding data required for debug drawing joint limits
+    Physics::CharacterPhysicsDebugDraw::JointDebugDrawData GetJointDebugDrawData(
+        const Physics::RagdollNodeConfiguration& ragdollNodeConfig,
+        const EMotionFX::ActorInstance* instance,
+        const AZStd::unordered_set<size_t>* cachedSelectedJointIndices)
+    {
+        Physics::CharacterPhysicsDebugDraw::JointDebugDrawData jointDebugDrawData;
+        const EMotionFX::Actor* actor = instance->GetActor();
+        const EMotionFX::Node* joint = actor->GetSkeleton()->FindNodeByName(ragdollNodeConfig.m_debugName.c_str());
+        if (!joint)
+        {
+            jointDebugDrawData.m_valid = false;
+            return jointDebugDrawData;
+        }
+
+        jointDebugDrawData.m_valid = true;
+        const size_t nodeIndex = joint->GetNodeIndex();
+        const bool jointSelected = cachedSelectedJointIndices &&
+            (cachedSelectedJointIndices->empty() || cachedSelectedJointIndices->find(nodeIndex) != cachedSelectedJointIndices->end());
+
+        if (!jointSelected)
+        {
+            jointDebugDrawData.m_visible = false;
+            return jointDebugDrawData;
+        }
+
+        const EMotionFX::Node* ragdollParentNode = instance->GetActor()->GetPhysicsSetup()->FindRagdollParentNode(joint);
+        if (!ragdollParentNode)
+        {
+            jointDebugDrawData.m_valid = false;
+            return jointDebugDrawData;
+        }
```

Comment:
should the validity check be done first?

With the current order the joint could be invalid (not having a parent node) but still be returned as valid and non-visible.

Reply:
> m_valid does actually default to false, so I think it should still be false in that case.

In this case it's true when it arrives to this code, it's set in line 79.

If being valid without a parent is ok while it's not-visible, then is not a problem. But I just wanted to bring it up in case other code just checks the valid boolean and assumes because it's true that it's got a parent node and crash.